packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1446 @@
+          enableAnimation: _enableAnimation,
+          isDiscrete: isDiscrete,
+          labelPainter: _startLabelPainter,
+          parentBox: this,
+          sliderTheme: _sliderTheme,
+          textDirection: _textDirection,
+          value: startValue,
+        );
+      }
+      if (_lastThumbSelection == Thumb.end) {
+        _sliderTheme.overlayShape.paint(
+          context,
+          endThumbCenter,
+          activationAnimation: _overlayAnimation,
+          enableAnimation: _enableAnimation,
+          isDiscrete: isDiscrete,
+          labelPainter: _endLabelPainter,
+          parentBox: this,
+          sliderTheme: _sliderTheme,
+          textDirection: _textDirection,
+          value: endValue,
+        );
+      }
+    }
+
+    if (isDiscrete) {
+      final double tickMarkWidth = _sliderTheme.rangeTickMarkShape.getPreferredSize(
+        isEnabled: isEnabled,
+        sliderTheme: _sliderTheme,
+      ).width;
+      final double adjustedTrackWidth = trackRect.width - tickMarkWidth;
+      // If the tick marks would be too dense, don't bother painting them.
+      if (adjustedTrackWidth / divisions >= 3.0 * tickMarkWidth) {
+        final double dy = trackRect.center.dy;
+        for (int i = 0; i <= divisions; i++) {
+          final double value = i / divisions;
+          // The ticks are mapped to be within the track, so the tick mark width
+          // must be subtracted from the track width.
+          final double dx = trackRect.left + value * adjustedTrackWidth + tickMarkWidth / 2;
+          final Offset tickMarkOffset = Offset(dx, dy);
+          _sliderTheme.rangeTickMarkShape.paint(
+            context,
+            tickMarkOffset,
+            parentBox: this,
+            sliderTheme: _sliderTheme,
+            enableAnimation: _enableAnimation,
+            textDirection: _textDirection,
+            startThumbCenter: startThumbCenter,
+            endThumbCenter: endThumbCenter,
+            isEnabled: isEnabled,
+          );
+        }
+      }
+    }
+
+    final double thumbDelta = (endThumbCenter.dx - startThumbCenter.dx).abs();
+
+    Thumb bottomThumb = _lastThumbSelection == Thumb.start ? Thumb.end : Thumb.start;
+    final Offset bottomThumbCenter = _lastThumbSelection == Thumb.start ? endThumbCenter : startThumbCenter;
+    final Offset topThumbCenter = _lastThumbSelection == Thumb.start ? startThumbCenter : endThumbCenter;
+    final TextPainter bottomLabelPainter = _lastThumbSelection == Thumb.start ? _endLabelPainter : _startLabelPainter;
+    final TextPainter topLabelPainter = _lastThumbSelection == Thumb.start ? _startLabelPainter : _endLabelPainter;
+    final double bottomValue = _lastThumbSelection == Thumb.start ? endValue : startValue;
+    final double topValue = _lastThumbSelection == Thumb.start ? startValue : endValue;
+
+    if (isEnabled && labels != null && !_valueIndicatorAnimation.isDismissed) {
+      if (showValueIndicator) {
```

Comment:
Combine with above `if` statement to avoid nesting.

Reply:
Done